### ISSUE TITLE
Clef: Got `inappropriate ioctl for device` when passing password via redirected stdin

### ISSUE THREAD

**Issue by cygnusv** [ethereum/go-ethereum#21086]
Hi,
There seems to be a bug in `golang.org/x/crypto/ssh/terminal` (https://github.com/golang/go/issues/19909) that [affects Clef](https://github.com/ethereum/go-ethereum/blob/7f95a85fd474b93e9f6441fa7f582f9bd089aeb7/signer/core/cliui.go#L66), which prevents from passing a password by redirected stdin.

Since there's no envvar or CLI flag that can be used as alternatives, it seems the only alternative is by using `--stdio-ui` and crafting a response with the password there, but the workaround seems excessive to me.

#### System information

Geth version: `geth 1.9.13-stable-cbc4ac26`
OS & Version: Linux

#### Expected behaviour
Clef decrypting the master seed file as usual.
#### Actual behaviour
```
> ERROR[05-14|00:44:24.154] Failed to read password                  err="inappropriate ioctl for device"
-----------------------
WARN [05-14|00:44:24.154] Failed to open master, rules disabled    err="inappropriate ioctl for device"
```

#### Steps to reproduce the behaviour
`$ echo -e "ok\n$CLEF_PASSWORD" | clef`

#### Backtrace

````
$ echo -e "ok\n$CLEF_PASSWORD" | clef

WARNING!

Clef is an account management tool. It may, like any software, contain bugs.

Please take care to
- backup your keystore files,
- verify that the keystore(s) can be opened with your password.

Clef is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
PURPOSE. See the GNU General Public License for more details.

Enter 'ok' to proceed:
> 
INFO [05-14|00:44:23.923] Using CLI as UI-channel 
INFO [05-14|00:44:24.154] Loaded 4byte database                    embeds=146841 locals=0 local=./4byte-custom.json
## Master Password

Please enter the password to decrypt the master seed
> ERROR[05-14|00:44:24.154] Failed to read password                  err="inappropriate ioctl for device"
-----------------------
WARN [05-14|00:44:24.154] Failed to open master, rules disabled    err="inappropriate ioctl for device"
````

**Comment by ligi** [comment]
I had the same problem @holiman fixed it here: https://github.com/ethereum/go-ethereum/pull/20960 -  but it is not merged yet. But you might just work on top of this PR as I do currently. Also might be merged soon - perhaps it can be part of the next PR triage which is next tuesday.

**Comment by ligi** [comment]
this PR was just merged to master - so closing this issue - please let us know if you still face this problem and feel free to reopen the issue in this case

**Comment by ligi** [comment]
fixed by @holiman via #20960
